modules/skybrowser/src/wwtcommunicator.cpp

```diff
@@ -0,0 +1,353 @@
+    }
+
+    void WwtCommunicator::setWebpageBorderColor(glm::ivec3 color) {
+        std::string stringColor = std::to_string(color.x) + ","
+            + std::to_string(color.y) + "," + std::to_string(color.z);
+        std::string script = "document.body.style.backgroundColor = 'rgb("
+            + stringColor + ")';";
+        executeJavascript(script);
+    }
+
+    void WwtCommunicator::setIsSyncedWithWwt(bool isSynced)
+    {
+        _isSyncedWithWwt = isSynced;
+    }
+
+    void WwtCommunicator::setEquatorialAim(const glm::dvec2& equatorial)
+    {
+        _equatorialAim = equatorial;
+        _equatorialAimIsDirty = true;
+    }
+
+    void WwtCommunicator::setBorderColor(const glm::ivec3& color)
+    {
+        _borderColor = color;
+        _borderColorIsDirty = true;
+    }
+
+    void WwtCommunicator::highlight(glm::ivec3 addition)
+    {
+        setWebpageBorderColor(_borderColor + addition);
+    }
+
+    void WwtCommunicator::removeHighlight(glm::ivec3 removal)
+    {
+        setWebpageBorderColor(_borderColor - removal);
+    }
+
+    void WwtCommunicator::updateBorderColor()
+    {
+        setWebpageBorderColor(_borderColor);
+    }
+
+    void WwtCommunicator::updateAim()
+    {
+        double roll = _isSyncedWithWwt ? skybrowser::cameraRoll() : 0.0;
+        // Message WorldWide Telescope current view
+        ghoul::Dictionary message = moveCamera(
+            _equatorialAim,
+            _verticalFov,
+            roll
+        );
+        sendMessageToWwt(message);
+    }
+
+    glm::dvec2 WwtCommunicator::fieldsOfView() {
+        glm::dvec2 browserFov = glm::dvec2(verticalFov() * browserRatio(), verticalFov());
+
+        return browserFov;
+    }
+
+    bool WwtCommunicator::hasLoadedImages() const {
+        return _hasLoadedImages;
+    }
+
+    glm::dvec2 WwtCommunicator::equatorialAim() const
+    {
+        return _equatorialAim;
+    }
+
+    void WwtCommunicator::setImageOrder(int i, int order) {
+        // Find in selected images list
+        auto current = std::find(
+            std::begin(_selectedImages),
+            std::end(_selectedImages),
+            i
+        );
+        auto target = std::begin(_selectedImages) + order;
+
+        // Make sure the image was found in the list
+        if (current != std::end(_selectedImages) && target != std::end(_selectedImages)) {
+            // Swap the two images
+            std::iter_swap(current, target);
+        }
+
+        int reverseOrder = _selectedImages.size() - order - 1;
+        ghoul::Dictionary message = setLayerOrder(std::to_string(i),
+            reverseOrder);
+        sendMessageToWwt(message);
+    }
+
+    void WwtCommunicator::loadImageCollection(const std::string& collection) {
+        sendMessageToWwt(loadCollection(collection));
+        _hasLoadedImages = true;
+    }
+
+    void WwtCommunicator::setImageOpacity(int i, float opacity) {
+        ghoul::Dictionary msg = setImageOpacity(std::to_string(i), opacity);
```

Comment:
In all of these functions that return a ghoul::Dictionary that is send as a message to WWT, it would be nice to reflect that in the function naming to make sure that it is immediately visible that the function doesn't actually **do** the thing, but rather it generates the message that will do the thing.